lang/c/examples/quickstop.c

```diff
@@ -25,214 +25,69 @@
+			avro_datum_decref(array_chunk);
+			fprintf(stdout, "QUICKSTOP: size[%lu]  : %s\n", avro_array_size(array_chunk), json);
+			free(json);
+			avro_datum_decref(array_chunk);
+		}
 	}
 
-	if (avro_file_writer_append_value(db, &person)) {
-		fprintf(stderr,
-			"Unable to write Person value to memory buffer\nMessage: %s\n", avro_strerror());
-		exit(EXIT_FAILURE);
-	}
+	fprintf(stdout, "\n================================================\n");
 
-	/* Decrement all our references to prevent memory from leaking */
-	avro_value_decref(&person);
-	avro_value_iface_decref(person_class);
 }
 
-int print_person(avro_file_reader_t db, avro_schema_t reader_schema)
+int process_pgd_data()
 {
-
-	avro_value_iface_t  *person_class =
-	    avro_generic_class_from_schema(person_schema);
-
-	avro_value_t person;
-	avro_generic_value_new(person_class, &person);
-
-	int rval;
-
-	rval = avro_file_reader_read_value(db, &person);
-	if (rval == 0) {
-		int64_t id;
-		int32_t age;
-		int32_t *p;
-		size_t size;
-		avro_value_t id_value;
-		avro_value_t first_value;
-		avro_value_t last_value;
-		avro_value_t age_value;
-		avro_value_t phone_value;
-
-		if (avro_value_get_by_name(&person, "ID", &id_value, NULL) == 0) {
-			avro_value_get_long(&id_value, &id);
-			fprintf(stdout, "%"PRId64" | ", id);
-		}
-		if (avro_value_get_by_name(&person, "First", &first_value, NULL) == 0) {
-			avro_value_get_string(&first_value, &p, &size);
-			fprintf(stdout, "%15s | ", p);
-		}
-		if (avro_value_get_by_name(&person, "Last", &last_value, NULL) == 0) {
-			avro_value_get_string(&last_value, &p, &size);
-			fprintf(stdout, "%15s | ", p);
-		}
-		if (avro_value_get_by_name(&person, "Phone", &phone_value, NULL) == 0) {
-			avro_value_get_string(&phone_value, &p, &size);
-			fprintf(stdout, "%15s | ", p);
-		}
-		if (avro_value_get_by_name(&person, "Age", &age_value, NULL) == 0) {
-			avro_value_get_int(&age_value, &age);
-			fprintf(stdout, "%"PRId32" | ", age);
-		}
-		fprintf(stdout, "\n");
-
-		/* We no longer need this memory */
-		avro_value_decref(&person);
-		avro_value_iface_decref(person_class);
-	}
-	return rval;
-}
-
-int main(void)
-{
-	int rval;
-	avro_file_reader_t dbreader;
-	avro_file_writer_t db;
-	avro_schema_t projection_schema, first_name_schema, phone_schema;
-	int64_t i;
-	const char *dbname = "quickstop.db";
-	char number[15] = {0};
-
-	/* Initialize the schema structure from JSON */
-	init_schema();
-
-	/* Delete the database if it exists */
-	remove(dbname);
-	/* Create a new database */
-	rval = avro_file_writer_create_with_codec
-	    (dbname, person_schema, &db, QUICKSTOP_CODEC, 0);
-	if (rval) {
-		fprintf(stderr, "There was an error creating %s\n", dbname);
-		fprintf(stderr, " error message: %s\n", avro_strerror());
-		exit(EXIT_FAILURE);
-	}
-
-	/* Add lots of people to the database */
-	for (i = 0; i < 1000; i++)
+	const char *avro_file = "f_sgeb25868e7855fe7a1d7f446f07923744_14d_v1609459200000_ts1609474609422.avro";
```

Comment:
This is a 60 mb avro data file. I read each of the arrays with 512 elements in the chunk. Without json-dump the peak value the program reached is 25 mb of res memory.